src/typecheck.c

```diff
@@ -617,10 +624,10 @@ static ExpressionTypes *typecheck_expression(TypeContext *ctx, const AstExpressi
         break;
     case AST_EXPR_NEG:
         result = typecheck_expression(ctx, &expr->data.operands[0])->type;
-        if (result->kind != TYPE_SIGNED_INTEGER && result->kind != TYPE_DOUBLE)
+        if (result->kind != TYPE_SIGNED_INTEGER && result->kind != TYPE_DOUBLE && result->kind != TYPE_FLOAT)
```

Comment:
Same thing, but simpler:
```suggestion
        if (result->kind != TYPE_SIGNED_INTEGER && !is_float_type(result))
```